sky/skylet/providers/aws/config.py

```diff
@@ -612,6 +671,38 @@ def _configure_subnet(config):
     return config
 
 
+def _get_vpc_id_by_name(vpc_name: str, config: Dict[str, Any]) -> str:
+    """Returns the VPC ID of the unique VPC with a given name.
+
+    Exits with code 1 if:
+      - No VPC with the given name is found in the current region.
+      - More than 1 VPC with the given name are found in the current region.
+    """
+    ec2 = _resource("ec2", config)
+    # Look in the "Name" tag (shown as Name column in console).
+    filters = [{"Name": "tag:Name", "Values": [vpc_name]}]
+    vpcs = [vpc for vpc in ec2.vpcs.filter(Filters=filters)]
+    if not vpcs:
+        logger.error(
+            f"ERROR_NO_NODES_LAUNCHED: No VPC with name {vpc_name!r} is found "
```

Comment:
nit: Can we add a prefix `SKYPILOT_` to avoid potential conflicts with other error codes?